src/main/java/uk/gov/cshr/service/VerificationCodeDeterminationService.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.cshr.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.cshr.domain.*;
+import uk.gov.cshr.exception.VerificationCodeTypeNotFound;
+
+@Service
+public class VerificationCodeDeterminationService {
+
+    private final EmailUpdateService emailUpdateService;
+
+    private final ReactivationService reactivationService;
+
+
+    public VerificationCodeDeterminationService(EmailUpdateService emailUpdateService,
+                                                ReactivationService reactivationService) {
+        this.emailUpdateService = emailUpdateService;
+        this.reactivationService = reactivationService;
+    }
+
+    public VerificationCodeDetermination getCodeType(String code) {
+        if (emailUpdateService.existsByCode(code)) {
```

Comment:
Should sign up be included as a code type?

Reply:
Im going to leave as is. Sign-up has a slightly different flow (and out of scope for this PR). It could be refactored out at a later stage, but it should be sufficient for now.